src/diffusers/models/attention_dispatch.py

```diff
@@ -1025,6 +1025,68 @@ def _all_to_all_single(x: torch.Tensor, group) -> torch.Tensor:
     x = _wait_tensor(x)
     return x
 
+def _all_to_all_dim_exchange(x: torch.Tensor, scatter_idx: int = 2, gather_idx: int = 1, group=None) -> torch.Tensor:
```

Comment:
Could we also add some basic docstrings to this function because it will help readability. Some commentaries on what the function is doing will also be helpful.